pkg/scd/store/cockroach/constraints.go

```diff
@@ -0,0 +1,291 @@
+package cockroach
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/interuss/dss/pkg/logging"
+	dssmodels "github.com/interuss/dss/pkg/models"
+	scdmodels "github.com/interuss/dss/pkg/scd/models"
+	dsssql "github.com/interuss/dss/pkg/sql"
+
+	"github.com/golang/geo/s2"
+	"github.com/lib/pq"
+	"go.uber.org/multierr"
+)
+
+var (
+	constraintFieldsWithIndices   [9]string
+	constraintFieldsWithPrefix    string
+	constraintFieldsWithoutPrefix string
+)
+
+func init() {
+	constraintFieldsWithIndices[0] = "id"
+	constraintFieldsWithIndices[1] = "owner"
+	constraintFieldsWithIndices[2] = "version"
+	constraintFieldsWithIndices[3] = "url"
+	constraintFieldsWithIndices[4] = "altitude_lower"
+	constraintFieldsWithIndices[5] = "altitude_upper"
+	constraintFieldsWithIndices[6] = "starts_at"
+	constraintFieldsWithIndices[7] = "ends_at"
+	constraintFieldsWithIndices[8] = "updated_at"
+
+	constraintFieldsWithoutPrefix = strings.Join(
+		constraintFieldsWithIndices[:], ",",
+	)
+
+	withPrefix := make([]string, 9)
+	for idx, field := range constraintFieldsWithIndices {
+		withPrefix[idx] = "scd_constraints." + field
+	}
+
+	constraintFieldsWithPrefix = strings.Join(
+		withPrefix[:], ",",
+	)
+}
+
+func (c *repo) fetchConstraints(ctx context.Context, q dsssql.Queryable, query string, args ...interface{}) ([]*scdmodels.Constraint, error) {
+	rows, err := q.QueryContext(ctx, query, args...)
+	if err != nil {
+		logging.Logger.Warn(fmt.Sprintf("fetchConstraints: error in QueryContext with %s", query))
```

Comment:
I would prefer contextual logging here, i.e.:
```
logger := logging.WithValuesFromContext(ctx, logging.Logger)
logger.Warn("error in issueing db query", zap.String("query", query), zap.Error(err)
```
However, in general, we should avoid logging in library code and leave it to outer layers to log rich error messages.

Reply:
Ah yes, this was an oversight; that logging was for debugging and I had intended to remove it.  Now done.